.github/workflows/generate_chapters.yml

```diff
@@ -2,17 +2,16 @@
 ## Custom Web Almanac GitHub action ##
 ######################################
 #
-# Updates timestamps, increments CSS, and then runs "npm run generate"
-# on any pushes to main for markdown, templates or generate scripts
+# This generates the chapters and tests the website
 #
-# Then it tests the website
+# For pushed to main it also updates timestamps and increments CSS.
+# Then we open a Pull Request for any changes generated. We do this
+# only on a push, rather than on pull request, and only on main to
+# avoid avoid merge conflicts acorss pull requests (and also because
+# that would require personal access tokens to be set up for pull
+# requests from forks).
 #
-# Then we open a Pull Request for any changes generated.
-#
-# We do it on a push, rather than on pull request, and only on main to
-# avoid needing GitHub Personal Access Tokens and to avoid merge conflicts
-#
-name: Generate Chapters on Push
+name: Generate Chapters
 on:
   workflow_dispatch:
   push:
```

Comment:
I will not limit it to only the `main` branch as contributors might want to work on feature branches in their forks and want to be confident that their change passes testes before creating a PR.

But then the job is limited to this repository only, so I am not too sure exactly how and when when want it to run.

Reply:
I thought you'd say that - so tried to cover that in the initial comment:

> I'd like to keep this check off from generic pushes, as it does take a few mins to run. Think it's find to only run on pull request when code should be ready to commit.

IMHO running this on EVERY push is a lot of processing potentially on unfinished code that may fail. I think sufficient to do this on opening a pull request when code should be "production ready".

It also means we can avoid that annoying doubling up of the Actions on pull requests like we get on the linting action:

![Double Actions](https://user-images.githubusercontent.com/10931297/89444393-8d370880-d749-11ea-80bf-95b092f6527f.png)

It is quite common on a lot of repos I work on to only run tests on pull requests only.

If people want to run this test locally then they can easily do this locally, before they even commit, by running the following:
- `pytest`
- `npm run generate`
- `npm run tests`
IMHO it's better to do that for code changes and the GitHub action should be a safety net to protect the code, to ensure tests pass before merging, rather than something developers would be depending on in their workflow.

Reply:
> But then the job is limited to this repository only, so I am not too sure exactly how and when when want it to run.

Oh that's a good point forgot about that. It was restricted to this repo to sop opening pull requests for those working on `main` branch in own repo.

I presume that means it won't run on pull requests so change this to this:

```yml
    if: (github.event_name == 'push' && github.repository == 'HTTPArchive/almanac.httparchive.org') || github.event.pull_request.head.repo.full_name == 'HTTPArchive/almanac.httparchive.org'
```

Any way to split that long line into two?

Reply:
> Any way to split that long line into two?

YAML allows multi-line strings, but I have never tried when evaluation conditionals in GH Actions context, so I am not sure how robust their parser is.

Reply:
Actually I've reverted this as according to [this](https://github.community/t/preventing-a-step-from-running-on-a-pr-originating-from-a-fork/18068/2?u=bazzadp) github.repository is set to the destination repo on pull requests so should be fine as it was.